src/pages/Facility/settings/devices/components/ServiceHistoryForm.tsx

```diff
@@ -0,0 +1,212 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { format } from "date-fns";
+import { CalendarIcon } from "lucide-react";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { z } from "zod";
+
+import { cn } from "@/lib/utils";
+
+import { Button } from "@/components/ui/button";
+import { Calendar } from "@/components/ui/calendar";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { Textarea } from "@/components/ui/textarea";
+
+import mutate from "@/Utils/request/mutate";
+import { ServiceHistory } from "@/types/device/device";
+import deviceApi from "@/types/device/deviceApi";
+
+interface ServiceHistoryFormProps {
+  facilityId: string;
+  deviceId: string;
+  serviceRecord?: ServiceHistory | null;
+  onOpenChange: (open: boolean) => void;
+}
+
+const formSchema = z.object({
+  note: z.string().min(1, { message: "Notes are required" }),
+  serviced_on: z.date({ required_error: "Service date is required" }),
+});
+
+type FormValues = z.infer<typeof formSchema>;
+
+export default function ServiceHistoryForm({
+  facilityId,
+  deviceId,
+  serviceRecord,
+  onOpenChange,
+}: ServiceHistoryFormProps) {
+  const { t } = useTranslation();
+  const queryClient = useQueryClient();
+
+  const form = useForm<FormValues>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      note: "",
+      serviced_on: new Date(),
+    },
+  });
+
+  useEffect(() => {
+    if (serviceRecord) {
+      form.reset({
+        note: serviceRecord.note,
+        serviced_on: new Date(serviceRecord.serviced_on),
+      });
+    } else {
+      form.reset({
+        note: "",
+        serviced_on: new Date(),
+      });
+    }
+  }, [serviceRecord, form]);
+
+  const createMutation = useMutation({
+    mutationFn: mutate(deviceApi.createserviceHistory, {
+      pathParams: {
+        facilityId,
+        deviceId,
+      },
+    }),
+    onError: (error) => {
```

Comment:
On error isn't needed (only needed in rare cases when backend error isn't clear enough)